components/api_request_helper/api_request_helper_unittest.cc

```diff
@@ -198,4 +211,47 @@ TEST_F(ApiRequestHelperUnitTest, EnableCache) {
               base::NullCallback(), true);
 }
 
+TEST_F(ApiRequestHelperUnitTest, JsonParsing) {
+  base::RunLoop run_loop;
+  SendMessage(base::BindRepeating(
+                  [](base::RunLoop* run_loop, const std::string& response) {
+                    EXPECT_EQ(" Hello there!", response);
+                    run_loop->Quit();
+                  },
+                  &run_loop),
+              "data: {\"completion\": \" Hello there!\", \"stop\": null}");
```

Comment:
This should expect a dictionary with completion and stop keys

Reply:
Revised the test to accept dictionary